python/ray/tests/test_failure.py

```diff
@@ -727,24 +728,44 @@ def test_warning_for_too_many_nested_tasks(shutdown_only):
     ray.init(num_cpus=num_cpus)
     p = init_error_pubsub()
 
+    remote_wait = Semaphore.remote(value=0)
+    nested_wait = Semaphore.remote(value=0)
+
+    ray.get([
+        remote_wait.locked.remote(),
+        nested_wait.locked.remote(),
+    ])
+
     @ray.remote
     def f():
         time.sleep(1000)
         return 1
 
     @ray.remote
-    def h():
-        time.sleep(1)
+    def h(nested_waits):
+        nested_wait.release.remote()
+        ray.get(nested_waits)
         ray.get(f.remote())
```

Comment:
should we do all of this logic within `f` to guarantee that there will be 3*n tasks running at once?

Reply:
I think all tasks won't be terminated anyway until `f` is done? because h ray.get on f, and g ray.get on h